webapp/channels/src/components/interactive_dialog/dialog_element/dialog_element.tsx

```diff
@@ -163,15 +163,21 @@ export default class DialogElement extends React.PureComponent<Props, State> {
                 textSettingMaxLength = maxLength || TEXTAREA_DEFAULT_MAX_LENGTH;
             }
 
-            const textValue = value as string;
+            let assertedValue;
```

Comment:
Would it be better to check the subtype filed for `number` and do a proper conversion in that case? I am just wondering if there are other strange cases.

Reply:
Thanks, it looks smart. I fixed this.

In fixed code, `typeof value === 'number'` may look extra. This condition is needed to set some value when `value` is `null`, because `value` can be `null` if default value is not set. I came up with `assertedValue = value as number || ''`, but I rejected because it looks strange.